mongodb-recreate-and-restore.txt

```diff
@@ -0,0 +1,125 @@
+This text document is for outlining the process necessary on a given
+environment to create the backup from OCP3, create the backup from OCP4, spin
+up a new instance of Mongo on OCP4 and then feed it data from the old backups
+before scaling it up and removing the old database.
+
+# OCP3 Backup
+
+To begin the process, we log into Openshift with the command copied from the
+web console:
+
+$ oc login ...
+
+Next, we make sure we are on the intended namespace by using:
+
+$ oc project <name of project>
+
+In this case, it is f047a2-dev. Once there, we list the available pods.
+
+$ oc get pods
+
+In previous deployments of MongoDB on Openshift, we did not have properly
+configured replicasets with leaders, and so any pod will be sufficient for
+taking a backup. In this case, we will assume one is called 'hcap-mongodb-0'.
+
+$ oc rsh hcap-mongodb-0
+
+Now that we've shelled into the pod on OCP3, we can use mongodump to create
+a copy of the logs, using credentials from the console, found at Secrets ->
+hcap-mongodb
+
+sh-4.2$ mongodump -u <admin-username> -p <admin-password> -o ocp3-mongo-dump
+
+Once this operation completes, it will create a `ocp3-mongo-dump` folder in the current
+directory, which should be /opt/app-root/src/. Exit the remote shell.
+
+In your working directory, the next step is to sync the dumped data onto the
+local machine using the following command.
+
+$ oc rsync hcap-mongodb-0:/opt/app-root/src/dump .
+
+And with that, we have moved all of the mongo data from OCP3 onto the local
+machine. Next, we need to spin up a new pod on OCP4 and restore it there.
+
+# New Mongo Pod on OCP4
+
+For this pod, we will be using a helm chart designed to enforce best
+practices for mongo deployments. Specific parameters are held in
+openshift/mongo.helm.yml, but we still need to configure helm to work on the
+local machine.
+
+Several options are documented here: https://helm.sh/docs/intro/install/
+
+Once helm is installed, we need to log into OCP4 with the creds provided
+by the web console, and then add the bitnami repo with the following command:
+
+$ oc login ...
+$ helm repo add bitnami https://charts.bitnami.com/bitnami
+
+Now that bitnami is registered, we can use their helm chart to build a new
+mongo deployment on Openshift. Make space by scaling down the existing
+hcap-mongodb StatefulSet to one pod, and then delete the associated PVC in
```

Comment:
Do we need to backup data before this step?

Reply:
I would prefer to make our backups one we've stopped using this statefulset for logging to avoid data loss. As far as I know, every PVC will hold the same data - we can consult the rest of the team to be sure.

Reply:
Ahh roger, do we run out of space if we don't take the PVC down?

Reply:
Yep, not having enough storage space was what got us into this situation in the first place. This should resolve it though.